File: solutions/julia/captains-log/1/captains-log.jl
using Random

random_planet() = rand(['D', 'H', 'J', 'K', 'L', 'M', 'N', 'R', 'T', 'Y'])

random_ship_registry_number() = "NCC-$(rand(1000:9999))"

random_stardate() = rand(41000.0:42000.0)

random_stardate_v2() = rand(41000.0:0.1:42000.0)

function pick_starships(starships, number_needed)
    Random.shuffle!(starships)
    return starships[1:number_needed]
end

File: solutions/r/captains-log/1/captains-log.R
library(stringr)

random_planet_class <- function(number_needed) {
  planet_class <- c("D", "H", "J", "K", "L", "M", "N", "R", "T", "Y")
  sample(planet_class, number_needed, replace = TRUE)
}

random_ship_registry_number <- function() {
  preffix <- "NCC-"
  paste0(preffix, str_flatten(sample(1000:9999, 1)))
}

shuffle_starships <- function(starships) {
  sample(starships)
}

random_stardate <- function() {
  runif(1, min = 41000.0 , max = 42000.0)
}
